tldw_Server_API/app/core/Chatbooks/chatbook_service.py

```
@@ -2370,6 +2441,71 @@ def preview_chatbook(self, file_path: str) -> tuple[ChatbookManifest | None, str
+
+        Validates the archive against the service export directory and returns
+        (job_id, download_url, file_path, file_size). Raises ExportError when
+        the archive is invalid or job persistence fails.
+        """
+        job_id = str(uuid4())
+        file_path = Path(output_path).resolve()
+        expected_base = Path(self.export_dir).resolve()
+        try:
+            file_path.relative_to(expected_base)
+        except ValueError:
+            raise ExportError("Export path validation failed") from None
```

Ruff (0.15.15)

[warning] Avoid specifying long messages outside the exception class

(TRY003)